core/templates/services/context.service.spec.ts

```diff
@@ -244,14 +319,59 @@ describe('Context service', () => {
       () => {
         expect(ecs.canAddOrEditComponents()).toBe(true);
       });
+
+    it('should correctly affirm that page contains reference to skills',
+      () => {
+        expect(ecs.getPageContext()).toBe('skill_editor');
+        expect(ecs.canEntityReferToSkills()).toBe(true);
```

Comment:
ditto: missing precheck

Reply:
Ditto: spyOn(urlService, 'getPathname').and.returnValue('/skill_editor/123'); in beforeEach is cause for this. so when getPageContext() is called, it sets the pageContext to skill_editor as per its pathname.